src/v0/destinations/customerio_audience/type.ts

```diff
@@ -63,33 +60,18 @@ export const CustomerIOConnectionConfigSchema = z
 
 export type CustomerIOConnectionConfig = z.infer<typeof CustomerIOConnectionConfigSchema>;
 
-const SegmentActionSchema = z.nativeEnum(SegmentAction);
-
-// Message type specific to CustomerIO
-export const CustomerIOMessageSchema = z
-  .object({
-    type: z.literal(MessageTypeSchema.enum.record),
-    action: SegmentActionSchema,
-    identifiers: z
-      .record(z.string(), z.union([z.string(), z.number()]))
-      .superRefine((identifiers, ctx) => {
-        if (Object.keys(identifiers).length === 0) {
-          ctx.addIssue({
-            code: z.ZodIssueCode.custom,
-            message: 'cannot be empty',
-          });
-        } else if (Object.keys(identifiers).length !== 1) {
-          ctx.addIssue({
-            code: z.ZodIssueCode.custom,
-            message: 'only one identifier is supported',
-          });
-        }
-      }),
-  })
-  .passthrough();
-
-export type SegmentActionType = z.infer<typeof SegmentActionSchema>;
-
+// Message type specific to CustomerIO, based on RudderRecordV2Schema
+export const CustomerIOMessageSchema = RudderRecordV2Schema.extend({
+  // Override the identifiers field with CustomerIO-specific validation
+  // For CustomerIO, identifiers is required and must contain exactly one identifier
+  identifiers: z
+    .record(z.string().min(1), z.union([z.string(), z.number()]))
+    .refine((ids) => Object.keys(ids).length === 1, {
```

Comment:
Mainly to make it clear what the refine function receives as its argument.
```suggestion
    .refine((identifiersObject) => Object.keys(identifiersObject).length === 1, {
```

Reply:
Done